ionoscloud/data_source_servers.go

```diff
@@ -287,6 +292,22 @@ func dataSourceServersRead(ctx context.Context, d *schema.ResourceData, meta int
 					}
 				}
 			}
+
+			// Labels logic
+			if server.Id == nil {
+				return diag.FromErr(fmt.Errorf("expected a valid server ID from the API but received nil instead"))
+			}
+			labelsResponse, apiResponse, err := client.LabelsApi.DatacentersServersLabelsGet(ctx, datacenterId.(string), *server.Id).Depth(1).Execute()
```

Comment:
this part is gonna be a function no? I see it's the same as below

Reply:
@cristiGuranIonos done in 586238e4619fe7d41d00334bfc01c629d834a759, I changed the design, now it is way cleaner than before.